### DISCUSSION TITLE
Deployment / Installation

### DISCUSSION THREAD

**Discussion by dojyaaa-n** [Kab1r/Google-Assistant-Discord-Bot#3] [Q&A] [answered]
Can you explain more specifically how to install the bot correctly, or rather, what needs to be done in the code?

**Comment by Kab1r** [comment, marked as answer]
# Installation

## Runtime
Supported Python Version is documented [here](https://github.com/KabirKwatra/Google-Assistant-Discord-Bot/blob/master/runtime.txt), 3.7 as of March 2020
[LXC / Docker runtime](https://github.com/KabirKwatra/Google-Assistant-Discord-Bot/blob/master/Dockerfile)) is also supported.

## Dependencies
The required [dependencies](https://github.com/KabirKwatra/Google-Assistant-Discord-Bot/blob/master/requirements.txt) can be installed using the following command. This automated at container build time if using LXC runtime.
`pip install -r requirements.txt`

## Environment Variables
- `DISCORD_TOKEN` (see any discord bot tutorial)
- `GA_CREDENTIALS` (#1)
- `GA_DEVICE_ID` (#1)
- `GA_DEVICE_MODEL_ID` (#1)
- `GA_TOKEN` (#1)

**Reply by Kab1r** [reply]
Heroku deployments using python buildpack is also supported.

**Comment by Om255** [comment]
How I Tokenize the bot

**Reply by Kab1r** [reply]
Discord provides tokens for bot applications

**Reply by Om255** [reply]
How i can't understand

**Reply by Om255** [reply]
Can my bot will be ready to go without giving Token it's can't possible

**Reply by Om255** [reply]
Apna Jo file De h use tokenize kaisa karu

**Reply by Om255** [reply]
How pls tell

**Reply by Om255** [reply]
Pls Kabir tell me pls

**Reply by Kab1r** [reply]
You don't need to tokenize anything, just store each as an environment variable
